cpp/src/barretenberg/stdlib/merkle_tree/hash.hpp

```diff
@@ -69,11 +57,7 @@ inline std::vector<barretenberg::fr> compute_tree_native(std::vector<barretenber
     while (layer.size() > 1) {
         std::vector<barretenberg::fr> next_layer(layer.size() / 2);
         for (size_t i = 0; i < next_layer.size(); ++i) {
-            if (plonk::SYSTEM_COMPOSER == ComposerType::PLOOKUP) {
-                next_layer[i] = crypto::pedersen_hash::lookup::hash_multiple({ layer[i * 2], layer[i * 2 + 1] });
-            } else {
-                next_layer[i] = crypto::pedersen_hash::hash_multiple({ layer[i * 2], layer[i * 2 + 1] });
-            }
+            next_layer[i] = crypto::pedersen_hash::lookup::hash_multiple({ layer[i * 2], layer[i * 2 + 1] });
```

Comment:
We are using the non-lookup implementations of these functions in a3-circuits (and therefore in a3-packages TS too) as we are _not_ using UltraPlonk. Maybe @suyash67 can chime in here.

Reply:
Oh, ok, but then I guess you're not using these funtions? Or do you reset `SYSTEM_COMPOSER` in your branch? It seems to me the simplest thing here if you want easy switching between the two would be `template <bool use_lookup>` on these functions and then to set some some a3-circuits global.

Reply:
We set USE_TURBO and then make calls to these from our C++ and then a3-packages makes calls to pedersen cbind directly. I guess we should duplicate the cbinds and have ultra+turbo versions?

Reply:
This work should eliminate my USE_TURBO hack, right? 🤞

Reply:
I believe it can. @codygunton if that is the intention then it should be removed from our CMakeFiles too. 

But for the time being we would need to template the functions on composer type like Cody mentioned and duplicate the pedersen cbinds currently change their functionality/output based on composer type.

Reply:
The conditional logic here was done so that the tests would pass if you switch composer from ultra to turbo or vice-versa. Otherwise, we'd have to manually fix the failing tests by calling the correct function.

Therefore, although we use both versions of Pedersen hash in A3, the stdlib handles the abstraction of when to use which one depending on a constant in composers. This isn't related to stdlib pedersen hash and commitment gadgets.

For now, this looks fine to me. We'll have to manually change these if/when we test things with different composers. As for Cody's suggestion of templated functions, that sounds good to me (whenever we implement it). Minor correction would be, instead of a boolean template parameter, we should parametrise it using an enum: `PedersenType::Lookup/FixedBase/SomethingElse`